weldx/transformations.py

```diff
@@ -1021,11 +1040,20 @@ def get_local_coordinate_system(
         path = nx.shortest_path(
             self.graph, coordinate_system_name, reference_system_name
         )
-        lcs = self.graph.edges[path[0], path[1]]["lcs"]
-        length_path = len(path) - 1
-        if length_path > 1:
-            for i in np.arange(1, length_path):
-                lcs = lcs + self.graph.edges[path[i], path[i + 1]]["lcs"]
+        path_edges = [edge for edge in zip(path[:-1], path[1:])]
+
+        time_union = self.time_union(path_edges)
+
+        lcs = self.graph.edges[path_edges[0]]["lcs"]
+
+        if time_union is not None:
+            lcs = lcs.interp_time(time_union)
+            for edge in path_edges[1:]:
+                lcs = lcs + self.graph.edges[edge]["lcs"].interp_time(time_union)
+        else:
+            for edge in path_edges[1:]:
+                lcs = lcs + self.graph.edges[edge]["lcs"]
```

Comment:
What do you think about `LocalCoordinateSystem.interp_time` simply returning `self` without any interpolation when `None` is passed for `time`? @vhirtham 

Than this could simply be
```python
lcs = lcs.interp_time(time_union)
for edge in path_edges[1:]:
    lcs = lcs + self.graph.edges[edge]["lcs"].interp_time(time_union)
```

I think this could simplify calls to `interp_time` in some other places as well.

Reply:
Nice idea. We should do that.

Reply:
Done @vhirtham 

Ready to merge on my part if CI passes :)

Reply:
Like the other resulting simplifications too :) --- Merge at will!